pkg/ruler/ruler.go

```diff
@@ -109,7 +109,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	cfg.ExternalURL.URL, _ = url.Parse("") // Must be non-nil
 	f.Var(&cfg.ExternalURL, "ruler.external.url", "URL of alerts return path.")
 	f.DurationVar(&cfg.EvaluationInterval, "ruler.evaluation-interval", 1*time.Minute, "How frequently to evaluate rules")
-	f.DurationVar(&cfg.EvaluationDelay, "ruler.evaluation-delay-duration", 0, "Duration to delay the evaluation of rules to ensure they underlying metrics have been pushed to cortex.")
+	f.DurationVar(&cfg.EvaluationDelay, "ruler.evaluation-delay-duration-deprecated", 0, "Deprecated. Duration to delay the evaluation of rules to ensure they underlying metrics have been pushed to cortex.")
```

Comment:
It's tricky. With the changes in this PR we're keeping the `evaluation_delay_duration` in the ruler YAML config, but we're not reading it anymore, so we're basically introducing a breaking change anyway (but more subtle because if you configured it, we ignore it).

I'm wondering if we should make it simple: introduce a breaking change (mentioning it in the CHANGELOG) and remove the `evaluation_delay_duration` at all from the ruler config.

Reply:
I agree that keeping the field around (for yaml), but silently ignoring is the worst combination. 

Alternative to breaking YAML config would be to use this value as a default value in overrides, if configured and  warn about it being deprecated, and removing after 2 releases.

Reply:
> Alternative to breaking YAML config would be to use this value as a default value in overrides, if configured and warn about it being deprecated, and removing after 2 releases.

Yes, but would require some refactoring in the limits overrides because when you read the setting for a user you don't know if that value is the default or not.

Reply:
> Yes, but would require some refactoring in the limits overrides because when you read the setting for a user you don't know if that value is the default or not.

I think we can set the default into `t.Cfg.LimitsConfig.EvaluationDelay` from `t.Cfg.Ruler.EvaluationDelay` (if set, and `t.Cfg.LimitsConfig.EvaluationDelay` is 0) before calling `overrides.SetDefaultLimitsForYAMLUnmarshalling` in `initRuntimeConfig`.

Reply:
I've implemented my suggestion.